make create relationship macros work with and without ctes

adamtwo opened the pull request:
resolves #

This PR resolves an issue where the adapter could not materialize tables and views with CTE and without CTE.

### Description

The PR changes `teradata__create_table_as` and `teradata__create_view_as` macros to accommodate models with CTE's and without CTE's.


### Checklist
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` with information about my change

Commit:
make create relationship macros work with and without ctes

Commit:
added changelog entry

adamtwo commented:
@dbeatty10 ,
I believe this PR addresses the CTE vs. non-CTE syntax in tables and views. I tested it manually with 4 models (a view with cte, a view without cte, a table with cte, a table without cte). 

I couldn't figure out if the adapter has any automated tests that I could extend?

dbeatty10 commented:
@adamtwo ,
I just added integration tests here: https://github.com/dbeatty10/dbt-teradata/pull/8

See the [README](https://github.com/dbeatty10/dbt-teradata/blob/dfe1b070b4908434bb35cf74e20711d704e2c853/test/README.md) within the test directory for instructions on running them. Pay particular attention to the "Troubleshooting" section at the very bottom. Let me know if you run into any issues.

I ran your proposed changes against the integration tests, and most of them look good. There are a few tests not able to pass yet, but I think they are unrelated to your changes. I reported https://github.com/dbeatty10/dbt-teradata/issues/7 and will submit a fix for that. 

Thank you for making this improvement for CTE vs. non-CTE syntax! I think this can be merged once you're ready.

Commit:
merged from dev/0.19.0

adamtwo commented:
Thanks for adding the DBT test suite. I'll need to learn how to extend it as Teradata Vantage has some corner cases that the DBT adapter test framework doesn't test, e.g. a view with vs. without CTE or a seed with a few thousand rows.
Ill explore a bit and try to include these tests in this PR.

Commit:
removed a need for stored procedures, implemented create/drop schema, added comprehensive tests

Commit:
updated changelog

adamtwo commented:
OK, I've ended up covering a lot of ground with this PR:
* I eliminated the need for `dbt_drop_relation_if_exists` stored procedure. I use the SQL hint syntax `/*+ IF EXISTS */` to inform `connections.py` that it should ignore specific exceptions. Since Teradata doesn't support hints, it treats `/*+ IF EXISTS */` as a regular comment and ignores it.
* Added tests for the table/view with/without CTE cases.
* Implemented `create_schema` and `drop_schema` macros.
* All these steps above simplify the test setup. All that is required right now is to have a vantage instance available, point to it, and run the tests. It's a prelude to automating our tests in GH Actions.

adamtwo committed:
Merge branch 'dev/0.19.0' into handle-tables-and-views-with-and-without-ctes

dbeatty10 commented:
Major wins in this PR!
- eliminating the `dbt_drop_relation_if_exists` stored procedure.
- custom tests for Teradata-specific behavior
- `create_schema` and `drop_schema macros`

Really sets things up perfectly for automating tests.

Incredible work, @adamtwo !
